webrender_api/src/webgl.rs

```diff
@@ -1040,4 +1044,13 @@ impl WebGLCommand {
         gl.shader_source(shader_id.get(), &[source.as_bytes()]);
         gl.compile_shader(shader_id.get());
     }
+
+    fn fence_and_wait_sync(gl: &gl::Gl) {
+        // Call FenceSync and ClientWaitSync to ensure that textures are ready.
+        let sync = gl.fence_sync(gl::SYNC_GPU_COMMANDS_COMPLETE, 0);
+        // SYNC_FLUSH_COMMANDS_BIT is used to automatically generate a glFlush before blocking on the sync object.
+        gl.client_wait_sync(sync, gl::SYNC_FLUSH_COMMANDS_BIT, gl::TIMEOUT_IGNORED);
```

Comment:
Why does this need to be a ```glClientWaitSync``` ? Wouldn't a ```glWaitSync``` be more appropriate in this case, so that the CPU isn't blocked? I would have though a ```glWaitSync``` would be enough to sync between the two command streams?